src/guidellm/backend/base.py

```diff
@@ -157,5 +155,14 @@ def model_tokenizer(self, model: str) -> Optional[str]:
         :return: The tokenizer for the model, or None if it cannot be created.
         :rtype: Optional[str]
         """
+        raise NotImplementedError
+
+
+@functools.lru_cache(maxsize=1)
+def _cachable_default_model(backend: Backend) -> str:
+    if models := backend.available_models():
+        logger.debug(f"Default model: {models[0]}")
+        return models[0]
 
-        pass
+    logger.error("No models available.")
+    raise ValueError("No models available.")
```

Comment:
Does the `lru_cache` work fine with `type: Backend`? I am not sure bit it caches only hashable types since it is just a dictionary under the hood.

Reply:
Double checked, yes, this will work with type: Backend. The default implementation for hashable on python classes is to return a unique identifier for the class